### PR TITLE
generate the model construct code to make things more easy

### PR DESCRIPTION
**Pull request by parai**
1. This could make things really easy, when a model is generated, just include the weights.h and call the API nnom_model_create, a model is created and compiled, no more need to rewrite the model according to keras.
2. But now only some of the layers are supported, need to update nnom_utils to support more, but that's easy as the framework is ready.
3. Remove the QFormat handing for input and output, as now output shift and bias shift is handled by nnom_utils.
4. Only support one input and one output.

### CODE REVIEW

File: scripts/nnom_utils.py [hunk 1]
```diff
@@ -290,19 +287,111 @@ def generate_model(model, x_test, name='weights.h'):
...
+            fp.write('\tif(NULL == layer) return NULL;\n')
+        else:
+            fp.write('\tnnom_layer_t* layer[%d];\n'%(ID+1))
+        fp.write('\n\tnew_model(&model);\n\n')
+        for _,(id,layer) in LI.items():
+            if(is_skipable_layer(layer)):
+                continue
+            if('input' in layer.name):
+                fp.write('\tlayer[%d] = Input(shape%s, nnom_input_data);\n'%(id,layer.input_shape[1:]))
+            elif('conv2d' in layer.name):
```
[the remark this answers is not in the document]
Comment (parai) [reply scripts/nnom_utils.py]: Hi @majianjia :
I would like to do that, but I didn't have a model use conv1d.

I create a very simple demo:
```python
x = Input(shape=(128,1))
conv1 = Conv1D(32, kernel_size=5, strides=1, padding='same')(x)
model = Model(inputs=x, outputs=conv1)
```
And could get below configuration:
{'name': 'conv1d_3', 'trainable': True, 'filters': 32, 'kernel_size': (5,), 'strides': (1,), 'padding': 'same', 'data_format': 'channels_last', 'dilation_rate': (1,), 'activation': 'linear', 'use_bias': True, 'kernel_initializer': {'class_name': 'VarianceScaling', 'config': {'scale': 1.0, 'mode': 'fan_avg', 'distribution': 'uniform', 'seed': None}}, 'bias_initializer': {'class_name': 'Zeros', 'config': {}}, 'kernel_regularizer': None, 'bias_regularizer': None, 'activity_regularizer': None, 'kernel_constraint': None, 'bias_constraint': None}

So it's simple that apply below patch to nnom_util.py would solve the problem:
```patch
diff --git a/scripts/nnom_utils.py b/scripts/nnom_utils.py
index affa07b..f6eaed1 100644
--- a/scripts/nnom_utils.py
+++ b/scripts/nnom_utils.py
@@ -345,6 +345,12 @@ def generate_model(model, x_test, name='weights.h'):
                 continue
             if('input' in layer.name):
                 fp.write('\tlayer[%d] = Input(shape%s, nnom_input_data);\n'%(id,layer.input_shape[1:]))
+            elif('conv1d' in layer.name):
+                inp = layer.input.name.replace(':','/').split('/')[0]
+                cfg = layer.get_config()
+                fp.write('\tlayer[{0}] = model.hook(Conv1D({1}, kernel({2},1), stride({3},1), PADDING_{4}, &{5}_w, &{5}_b), layer[{6}]);\n'.format(
+                    id, layer.output.shape[-1], cfg['kernel_size'][0], cfg['strides'][0], cfg['padding'].upper(),
+                    layer.name, LI[inp][0]));
             elif('conv2d' in layer.name):
                 inp = layer.input.name.replace(':','/').split('/')[0]
                 cfg = layer.get_config()
```

Could you help to have a test on it?

[the remark this answers is not in the document]
Comment (parai) [reply scripts/nnom_utils.py]: add the commit https://github.com/majianjia/nnom/pull/11/commits/dabf22f3a75efdc046ed5829b75d14b8c5236129, but please help to do verify.

[the remark this answers is not in the document]
Comment (majianjia) [reply scripts/nnom_utils.py]: Thanks, it is nearly correct, only the parameter order in kernel(), and strides() need to be changed a bit. 
Please change to this:
'Conv1D({1}, kernel(1, {2}), stride(1, {3}), PADDING_{4}, &{5}_w, &{5}_b)'
